packages/clay-core/src/table/Cell.tsx

```diff
@@ -110,31 +126,40 @@ export const Column = React.forwardRef<HTMLTableCellElement, Props>(
 					'table-cell-ws-nowrap': !wrap,
 					'table-head-title': isHead,
 				})}
+				data-id={
+					typeof keyValue === 'number'
+						? `number,${keyValue}`
+						: `string,${keyValue}`
+				}
+				onClick={(event) => {
+					if (!(isHead && sortable)) {
+						return;
+					}
+
+					event.preventDefault();
+					onSortChange(
+						{
+							column: keyValue!,
+							direction:
+								sort && keyValue === sort.column
+									? sort.direction === 'ascending'
+										? 'descending'
+										: 'ascending'
+									: 'ascending',
+						},
+						textValue!
+					);
+				}}
 				ref={ref}
+				role={treegrid ? 'gridcell' : undefined}
 			>
 				{isHead && sortable ? (
 					<a
-						aria-describedby={
-							isHead && sortable ? sortDescriptionId : undefined
-						}
 						className="inline-item text-truncate-inline"
 						href="#"
-						onClick={(event) => {
-							event.preventDefault();
-							onSortChange(
-								{
-									column: keyValue!,
-									direction:
-										sort && keyValue === sort.column
-											? sort.direction === 'ascending'
-												? 'descending'
-												: 'ascending'
-											: 'ascending',
-								},
-								textValue!
-							);
-						}}
-						role="button"
+						onClick={(event) => event.preventDefault()}
+						role="presentation"
+						tabIndex={-1}
```

Comment:
I made some changes here, the Link at this moment is only serving as the hover interaction when hovering the mouse over it to cover this Lexicon behavior but the sortable action is done when clicking on the column header to better deal with accessibility and SR assertiveness .

The idea is to remove this link and add the appropriate style without needing a link